### ISSUE TITLE
Upstream FindBotan.cmake to CMake project

### ISSUE THREAD

**Issue by securitykernel** [rnpgp/rnp#2121]
It has been a long-term issue that CMake does not ship with a `FindBotan.cmake` find module to search for and link against a Botan installation. People start writing their own find modules again and again, that basically do the same, which you can see [GitHub search](https://github.com/search?q=findbotan.cmake&type=code), for example. For existing but also new users of Botan, it would be great if they had out-of-the-box support for Botan in CMake projects.

I found that the [`FindBotan`](https://github.com/rnpgp/rnp/blob/main/cmake/Modules/FindBotan.cmake) script you ship with rnp looks the most mature and already supports Botan 2.x and 3.x. Would it be possible to upstream it to the [CMake project](https://gitlab.kitware.com/cmake/cmake/-/tree/master/Modules)? The only requirement from CMake's side would be to find a [dedicated module maintainer](https://gitlab.kitware.com/cmake/community/-/wikis/doc/cmake/dev/Module-Maintainers). If none of you can do this, I would be willing to step in to make this happen.

/cc @randombit @reneme @lieser

**Comment by ni4** [comment]
@ronaldtse what do you think?
We already have separate repository for CMake modules, used in rnp: https://github.com/rnpgp/cmake-modules/
Actually never know/think before about upstreaming CMake modules to the CMake repository, but the idea itself looks good to me.

**Comment by securitykernel** [comment]
Any update on this?

**Comment by ni4** [comment]
@securitykernel Sorry for a delay. I absolutely don't mind moving it to upstream/CMake. However let's wait for @ronaldtse reply.

**Comment by securitykernel** [comment]
Sure, sounds fine!

**Comment by ronaldtse** [comment]
@securitykernel thank you for the suggestion! We'd be happy to contribute it upstream. Would you be willing to jointly maintain the module upstream?

**Comment by securitykernel** [comment]
Sure, let's do it!

**Comment by securitykernel** [comment]
Asked the CMake maintainers if it would be accepted. Waiting for their response. Meanwhile, I started to integrate it into their repo. Are you fine with re-licensing it to BSD 3-Clause License, as this is the [default](https://cmake.org/cmake/help/git-master/manual/cmake-developer.7.html#a-sample-find-module) for all find modules shipped with CMake?

**Comment by ronaldtse** [comment]
@securitykernel yes we would be willing to re-license the module to BSD 3-clause -- BSD 2-clause is more general than BSD 3-clause so it is fine.

**Comment by securitykernel** [comment]
So we got everything right now, I think. Can you please test that https://github.com/randombit/botan/pull/3722 works for you? That would give us a bit of more confirmation.
